### PR TITLE
280 - Improve file name for google drive

### PR DESCRIPTION
**Pull request by Pbarbosa92**
This PR enhances the file naming mechanism for downloads.

Previously, the filename was simply derived by splitting the URL and using the last segment. Now, the process has been refined to extract the filename from the `Content-Disposition` header in the HTTP response.

Additionally, we introduced several quality-of-life improvements, including a progress bar for both the download and extraction processes.

Issue https://github.com/inductiva/tasks/issues/280

### CODE REVIEW

**Comment by Pbarbosa92** [comment]
Download:
Image

Unzip:
Image

File: inductiva/utils/files.py [hunk 1]
```diff
@@ -136,21 +166,37 @@ def download_from_url(url: str, unzip: bool = False) -> str:
...
+        #Get head to check the filename
+        filename = url.split("/")[-1]
+        response = requests.head(url, allow_redirects=True, timeout=3)
+
+        # Check if the 'Content-Disposition' header is present
+        content_disposition = response.headers.get("Content-Disposition")
+        if content_disposition and "filename=" in content_disposition:
+            filename = content_disposition.split("filename=")[1].strip('"')
+
+        logging.info("Downloading from URL to the local path: %s", filename)
```
Comment (sergiomsantos) [inline inductiva/utils/files.py:179]: This is the main action that this function performs. We can prefix it with a bullet:

```suggestion
        logging.info("■ Downloading from URL to %s", filename)
```

File: inductiva/utils/files.py [hunk 2]
```diff
@@ -136,21 +166,37 @@ def download_from_url(url: str, unzip: bool = False) -> str:
...
         raise url_error
 
+    resulting_path = downloaded_to
+
     # Unzip all files as they were zipped.
     if unzip and zipfile.is_zipfile(downloaded_to):
-        local_path = local_path.with_suffix("")
-        logging.info("■ Uncompressing the downloaded file to: %s", local_path)
+        resulting_path = pathlib.Path(downloaded_to).with_suffix("")
+        logging.info("■ Uncompressing the downloaded file to: %s",
```
Comment (sergiomsantos) [inline inductiva/utils/files.py:196]:
```suggestion
        logging.info("■ Uncompressing the downloaded archive to %s",
```

File: inductiva/utils/files.py [hunk 3]
```diff
@@ -118,11 +121,38 @@ def _unzip(zip_path: pathlib.Path):
...
+            Size of each block (in tqdm units) [default: 1].
+        tsize  : int, optional
+            Total size (in tqdm units). If [default: None] remains unchanged.
+        """
+        if tsize is not None:
+            t.total = tsize
+        t.update((b - last_b[0]) * bsize)
+        last_b[0] = b
+
+    return update_to
```
Comment (sergiomsantos) [inline inductiva/utils/files.py:153]:
```suggestion
def tqdm_refreshhook(tqdm_bar):
    def update_to(blocks, block_size, total_size):
        if total_size is not None:
            tqdm_bar.total = total_size
        tqdm_bar.n = blocks*block_size
        tqdm_bar.refresh()

    return update_to
```

File: inductiva/utils/files.py [hunk 4]
```diff
@@ -136,21 +166,37 @@ def download_from_url(url: str, unzip: bool = False) -> str:
...
+            filename = content_disposition.split("filename=")[1].strip('"')
+
+        logging.info("Downloading from URL to the local path: %s", filename)
+        with tqdm(unit="B",
+                  unit_scale=True,
+                  unit_divisor=1024,
+                  miniters=1,
+                  desc=filename) as t:
+            downloaded_to, _ = urllib.request.urlretrieve(
+                url, filename=pathlib.Path(filename), reporthook=my_hook(t))
```
Comment (sergiomsantos) [inline inductiva/utils/files.py:186]:
```suggestion
                url, filename=pathlib.Path(filename), reporthook= tqdm_refreshhook(t))
```

File: inductiva/utils/files.py [hunk 5]
```diff
@@ -136,21 +166,37 @@ def download_from_url(url: str, unzip: bool = False) -> str:
...
+
     # Unzip all files as they were zipped.
     if unzip and zipfile.is_zipfile(downloaded_to):
-        local_path = local_path.with_suffix("")
-        logging.info("■ Uncompressing the downloaded file to: %s", local_path)
+        resulting_path = pathlib.Path(downloaded_to).with_suffix("")
+        logging.info("■ Uncompressing the downloaded file to: %s",
+                     resulting_path)
         _unzip(downloaded_to)
+        #remove .zip extension
```
Comment (sergiomsantos) [inline inductiva/utils/files.py:199]: misplaced comment

**Review by sergiomsantos — COMMENTED** [review]
just some simplification suggestions

**Comment by Pbarbosa92** [comment]
Can you QA this PR @sergiomsantos ?

**Comment by sergiomsantos** [comment]
@Pbarbosa92 Looking good.

**Pull request merged**